### PR TITLE
MCH: initial implementation of ROF analysis

### PR DESCRIPTION
**Pull request by aferrero2707**

### CODE REVIEW

**Comment by aferrero2707** [comment]
@aphecetche could you have a look at the simplified code, and let me know if it is ok? Thanks!

File: Modules/MUON/MCH/src/PhysicsTaskRofs.cxx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+void PhysicsTaskRofs::startOfCycle()
+{
+  ILOG(Info, Support) << "startOfCycle" << ENDM;
+}
+
+void PhysicsTaskRofs::plotROF(const o2::mch::ROFRecord& rof, gsl::span<const o2::mch::Digit> digits)
+{
+  std::array<bool, 10> stations{ false };
```
Comment (aphecetche) [inline Modules/MUON/MCH/src/PhysicsTaskRofs.cxx:122]: should be 5 not 10 for stations

File: Modules/MUON/MCH/src/PhysicsTaskRofs.cxx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+#include "Framework/WorkflowSpec.h"
+#include "Framework/DataRefUtils.h"
+
+#include "MCHRawElecMap/Mapper.h"
+#include "MCHMappingInterface/Segmentation.h"
+#include "MCHRawDecoder/PageDecoder.h"
+#include "MCHRawDecoder/ErrorCodes.h"
+#include "MCHBase/DecoderError.h"
+
+#define publishObject(object, drawOption, statBox)        \
```
Comment (aphecetche) [inline Modules/MUON/MCH/src/PhysicsTaskRofs.cxx:33]: macro could / should be changed to function

```
template <typename T>
  void publishObject(std::shared_ptr<T> histo, const char* drawOption, bool statBox)
  {
    histo->SetOption(drawOption);
    if (!statBox) {
      histo->SetStats(0);
    }
    mAllHistograms.push_back(histo.get());
    getObjectsManager()->startPublishing(histo.get());
  }
```

File: Modules/MUON/MCH/include/MCH/PhysicsTaskRofs.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  void endOfCycle() override;
+  void endOfActivity(Activity& activity) override;
+  void reset() override;
+
+ private:
+  void plotROF(const o2::mch::ROFRecord& rof, gsl::span<const o2::mch::Digit> digits);
+  /// \brief helper function for storing the histograms to a ROOT file on disk
+  void writeHistos();
+
+  bool mSaveToRootFile{ false }; ///< flag for saving the histograms to a local ROOT file
```
Comment (aphecetche) [inline Modules/MUON/MCH/include/MCH/PhysicsTaskRofs.h:63]: why not using directly the QC option "saveObjectsToFile" ?

File: Modules/MUON/MCH/src/PhysicsTaskRofs.cxx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+  // ROF width distributions
+  mHistRofWidth = std::make_shared<TH1F>("RofWidth", "ROF width", 5000 / 25, 0, 5000 / 25);
+  publishObject(mHistRofWidth, "hist", true);
+
+  // Number of stations per ROF
+  mHistRofNStations = std::make_shared<TH1F>("RofNStations", "Number of stations per ROF", 7, 0, 7);
+  publishObject(mHistRofNStations, "hist", true);
+
+  // ROF time distribution
+  mHistRofTime = std::make_shared<TH1F>("RofTime", "ROF time distribution", 128 * 3600, 0, 128 * 3600);
```
Comment (aphecetche) [inline Modules/MUON/MCH/src/PhysicsTaskRofs.cxx:102]: should not use hardcoded constants here

**Comment by Barthelemy** [comment]
ok, we will merge it after the tests are ok (apart from macos broken test)

**Comment by aferrero2707** [comment]
@Barthelemy @knopers8 looks like all is green apart the macOS build, could you merge whenever possible? Thanks!

**Pull request merged**